Code/Mantid/Framework/PythonInterface/plugins/algorithms/DNSMergeRuns.py

```diff
@@ -0,0 +1,285 @@
+        self.declareProperty("Normalize", False, "If checked, the merged data will be normalized, "
+                             "otherwise the separate normalization workspace will be created.")
+        return
+
+    def _dimensions_valid(self):
+        """
+        Checks validity of the workspace dimensions:
+        all given workspaces must have the same number of dimensions
+        and the same number of histograms
+        and the same number of bins
+        """
+        ndims = []
+        nhists = []
+        nbins = []
+        for wsname in self.workspace_names:
+            ws = api.AnalysisDataService.retrieve(wsname)
+            ndims.append(ws.getNumDims())
+            nhists.append(ws.getNumberHistograms())
+            nbins.append(ws.blocksize())
+
+        nd = ndims[0]
+        nh = nhists[0]
+        nb = nbins[0]
+        if ndims.count(nd) == len(ndims) and nhists.count(nh) == len(nhists) and nbins.count(nb) == len(nbins):
+            return True
+        else:
+            message = "Cannot merge workspaces with different dimensions."
+            self.log().error(message)
+            return False
+
+    def _ws_exist(self):
+        """
+        Checks whether the workspace and its normalization workspaces exist
+        """
+        for wsname in self.workspace_names:
+            if not api.AnalysisDataService.doesExist(wsname):
+                message = "Workspace " + wsname + " does not exist! Cannot merge."
+                self.log().error(message)
+                return False
+            if not api.AnalysisDataService.doesExist(wsname + '_NORM'):
+                message = "Workspace " + wsname + "_NORM does not exist! Cannot merge."
+                self.log().error(message)
+                return False
+
+        return True
+
+    def _can_merge(self):
+        """
+        checks whether it is possible to merge the given list of workspaces
+        """
+        # list of workspaces must not be empty
+        if not self.workspace_names:
+            message = "No workspace names has been specified! Nothing to merge."
+            self.log().error(message)
+            return False
+        # workspaces and normalization workspaces must exist
+        if not self._ws_exist():
+            return False
+        # they must have the same dimensions
+        if not self._dimensions_valid():
+            return False
+        # and the same wavelength
+        if not self._same_wavelength():
+            return False
+        # algorithm must warn if some properties_to_compare are different
+        ws1 = api.AnalysisDataService.retrieve(self.workspace_names[0])
+        run1 = ws1.getRun()
+        for wsname in self.workspace_names[1:]:
+            ws = api.AnalysisDataService.retrieve(wsname)
+            run = ws.getRun()
+            self._check_properties(run1, run)
+        return True
+
+    def _check_properties(self, lhs_run, rhs_run):
```

Comment:
To avoid code duplication the three algorithms that share this method can be inherited from an intermediate class:

```python
DNSAlgorithm(PythonAlgorithm):
    def _check_properties(self, lhs_run, rhs_run):
        ...

class DNSMergeRuns(DNSAlgorithm):
    def PyExec(self):
        ....
        self._check_properties(...)

class DNSFlippingRatioCorr(DNSAlgorithm):
    def PyExec(self):
        ....
        self._check_properties(...)

class DNSDetEffCorrVana(DNSAlgorithm):
    def PyExec(self):
        ....
        self._check_properties(...)

```